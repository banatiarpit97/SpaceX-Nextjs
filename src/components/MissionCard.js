import React from 'react';
import styles from './../styles/MissionCard.module.css';

const MissionCard = ({
    mission: {links, mission_id, mission_name, flight_number, launch_year, launch_success, launch_landing}
}) => (
    <section className={styles.container}>
        <img
            src={links.mission_patch}
            alt={mission_name}
            className={styles.image}
        />
        <h3 className={styles.name}>{mission_name} #{flight_number}</h3>
        {
            !!mission_id?.length && (
                <>
                    <h4 className={styles.type}>Mission Ids:</h4>
                    <ul className={styles.id_list}>
                        {mission_id.map(id => (
                            <li key={id} className={styles.detail}>{id}</li>
                        ))}
                    </ul>
                </>
            )
        }
        <div className={styles.flex}>
            <h4 className={styles.type}>Launch Year:</h4>
            <p className={styles.detail}>{launch_year}</p>
        </div>
        <div className={styles.flex}>
            <h4 className={styles.type}>Successful Launch:</h4>
            <p className={styles.detail}>{String(launch_success)}</p>
        </div>
        <div className={styles.flex}>
            <h4 className={styles.type}>Successful Landing:</h4>
            <p className={styles.detail}>{String(!!launch_landing)}</p>
        </div>
    </section>
);

export default MissionCard;