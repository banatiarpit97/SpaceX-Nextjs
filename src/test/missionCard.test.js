import React from 'react';
import renderer from 'react-test-renderer';

import MissionCard from './../components/MissionCard';

it('renders correctly', () => {
  const tree = renderer.create(<MissionCard />).toJSON();
  expect(tree).toMatchSnapshot();
});