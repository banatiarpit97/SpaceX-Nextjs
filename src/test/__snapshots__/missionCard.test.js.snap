// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="container"
>
  <img
    className="image"
  />
  <h3
    className="name"
  >
     #
  </h3>
  <div
    className="flex"
  >
    <h4
      className="type"
    >
      Launch Year:
    </h4>
    <p
      className="detail"
    />
  </div>
  <div
    className="flex"
  >
    <h4
      className="type"
    >
      Successful Launch:
    </h4>
    <p
      className="detail"
    >
      undefined
    </p>
  </div>
  <div
    className="flex"
  >
    <h4
      className="type"
    >
      Successful Landing:
    </h4>
    <p
      className="detail"
    >
      false
    </p>
  </div>
</section>
`;
